File: ServeUp-Client/ios/RNEventEmitter.m
//
//  RCTEventEmitter.m
//  myapp
//
//  Created by Yuval on 10/12/2023.
//
//  RCTEventEmitter.m

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE (RNEventEmitter, RCTEventEmitter)
    RCT_EXTERN_METHOD(supportedEvents)
@end

File: ServeUp-Client/ios/RNEventEmitter.swift
//
//  RNEventEmitter.swift
//  myapp
//
//  Created by Yuval on 10/12/2023.
//
//  RNEventEmitter.swift

import Foundation
import React

@objc(RNEventEmitter)
open class RNEventEmitter :RCTEventEmitter{
  
  public static var emitter: RCTEventEmitter!
  
  override public init() {
    super .init()
    RNEventEmitter.emitter=self
  }
  
  open override func supportedEvents() -> [String]! {
    ["onSent"]
  }
  @objc public override static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
